tobac/tracking.py

```diff
@@ -270,22 +341,35 @@ def linking_trackpy(
             memory=memory,
             # pos_columns=["hdim_1", "hdim_2"], # not working atm
             t_column="frame",
-            neighbor_strategy="KDTree",
+            neighbor_strategy=neighbor_strategy,
             link_strategy="auto",
             adaptive_step=adaptive_step,
-            adaptive_stop=adaptive_stop
+            adaptive_stop=adaptive_stop,
+            # dist_func=dist_func
             #                                 copy_features=False, diagnostics=False,
             #                                 hash_size=None, box_size=None, verify_integrity=True,
             #                                 retain_index=False
         )
         # recreate a single dataframe from the list
+
         trajectories_unfiltered = pd.concat(trajectories_unfiltered)
 
         # change to column names back
-        trajectories_unfiltered.rename(
-            columns={"y": "hdim_1", "x": "hdim_2"}, inplace=True
-        )
-        features.rename(columns={"y": "hdim_1", "x": "hdim_2"}, inplace=True)
+        if not is_3D:
+            trajectories_unfiltered.rename(
+                columns={"y": "hdim_1", "x": "hdim_2"}, inplace=True
+            )
+            features_linking.rename(
```

Comment:
features_linking doesn't necessarily need the columns renamed as it shouldn't be returned by the function

Reply:
I believe that this is resolved by the merge conflict change